craft_parts/state_manager/step_state.py

```diff
@@ -46,6 +46,7 @@ class MigrationState(BaseModel):
     files from shared areas on step cleanup.
     """
 
+    partition: str | None = None
```

Comment:
Tracking the partition name in the state to know where to get the content from when `contents()` is called.